wapitiCore/definitions/backup.py

```diff
@@ -18,38 +19,55 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
-TYPE = "vulnerability"
-
-NAME = "Backup file"
-SHORT_NAME = NAME
-
-WSTG_CODE = ["WSTG-CONF-04"]
-
-DESCRIPTION = (
-    "It may be possible to find backup files of scripts on the webserver that the web-admin put here "
-    "to save a previous version or backup files that are automatically generated by the software editor used "
-    "(like for example Emacs)."
-) + " " + (
-    "These copies may reveal interesting information like source code or credentials."
-)
-
-SOLUTION = (
-    "The webadmin must manually delete the backup files or remove it from the web root."
-) + " " + (
-    "He should also reconfigure its editor to deactivate automatic backups."
-)
-
-REFERENCES = [
-    {
-        "title": "OWASP: Review Old Backup and Unreferenced Files for Sensitive Information",
-        "url": (
-            "https://owasp.org/www-project-web-security-testing-guide/stable/4-Web_Application_Security_Testing/"
-            "02-Configuration_and_Deployment_Management_Testing/"
-            "04-Review_Old_Backup_and_Unreferenced_Files_for_Sensitive_Information.html"
+from wapitiCore.definitions.base import FindingBase
+
+
+class BackupFinding(FindingBase):
+    @classmethod
```

Comment:
what about using `@property` decorator  for all of these ?

Reply:
Not possible, we need abstract method for type enforcing. Python allowed `@property` on top of `@classmethod` at some point but removed it in recent versions of Python.

That's why we have to use methods.

Reply:
From Python documentation:

> Changed in version 3.11: Class methods can no longer wrap other descriptors such as property()